File: docs/docs/api-reference/iam/api_user/v1/examples/go/create-api-user.go
package main

import (
	"context"
	"log"

	api_userv1 "github.com/meshtrade/api/go/iam/api_user/v1"
	rolev1 "github.com/meshtrade/api/go/iam/role/v1"
)

func main() {
	// Create client (see client-setup.go for details)
	client, err := api_userv1.NewApiUserServiceGRPCClient(
		api_userv1.WithAddress("localhost", 8080),
		api_userv1.WithTLS(false),
		api_userv1.WithAPIKey("your-api-key"),
		api_userv1.WithGroup("your-group-id"),
	)
	if err != nil {
		log.Fatalf("Failed to create client: %v", err)
	}
	defer client.Close()

	// Prepare API user configuration
	apiUserToCreate := &api_userv1.APIUser{
		Owner:       "groups/your-group-id",
		DisplayName: "My API User",
		Roles: []rolev1.Role{
			rolev1.Role_ROLE_IAM_ADMIN,
		},
	}

	// Create the new API user
	apiUser, err := client.CreateApiUser(
		context.Background(),
		&api_userv1.CreateApiUserRequest{
			ApiUser: apiUserToCreate,
		},
	)
	if err != nil {
		log.Fatalf("Failed to create API user: %v", err)
	}

	log.Printf("Created API user: %s", apiUser.Name)
	log.Printf("Display name: %s", apiUser.DisplayName)
	log.Printf("API key: %s", apiUser.ApiKey) // Only shown on creation
}

File: docs/docs/api-reference/iam/api_user/v1/examples/python/create-api-user.py
import grpc

from meshtrade.iam.api_user.v1 import api_user_pb2, service_pb2, service_pb2_grpc
from meshtrade.option.v1 import role_pb2


def create_client_and_metadata():
    """Create client and authentication metadata."""
    channel = grpc.insecure_channel("localhost:8080")
    client = service_pb2_grpc.ApiUserServiceStub(channel)
    metadata = [
        ("authorization", "Bearer your-api-key"),
        ("x-group-id", "your-group-id"),
    ]
    return client, channel, metadata


def main():
    # Create client and metadata
    client, channel, metadata = create_client_and_metadata()

    try:
        # Create API user configuration
        api_user = api_user_pb2.APIUser(owner="groups/your-group-id", display_name="My API User", roles=[role_pb2.Role.ROLE_IAM_ADMIN])

        # Create request
        request = service_pb2.CreateApiUserRequest(api_user=api_user)

        # Make the API call
        response = client.CreateApiUser(request, metadata=metadata)

        print(f"Created API user: {response.name}")
        print(f"Display name: {response.display_name}")
        print(f"API key: {response.api_key}")  # Only shown on creation

    except grpc.RpcError as e:
        print(f"Failed to create API user: {e}")

    finally:
        channel.close()


if __name__ == "__main__":
    main()
